go/mysql/query.go

```diff
@@ -441,8 +442,7 @@ func (c *Conn) ReadQueryResult(maxrows int, wantfields bool) (*sqltypes.Result,
 				more = (statusFlags & ServerMoreResultsExists) != 0
 				result.StatusFlags = statusFlags
 			} else {
-				packetOk, err := c.parseOKPacket(data)
-				if err != nil {
+				if err := c.parseOKPacket(&packetOk, data); err != nil {
```

Comment:
Since we reuse `packetOk` here, is there anything from the original `c.readComQueryResponse` that could leak into this? Do we know all things are properly reset inside `parseOKPacket`?

Reply:
Yes they're properly reset but just to make sure this doesn't regress in the future, I've switched this call to use a brand-new stack-allocated packet. That way the API cannot possibly fail.